### PR TITLE
As/config method

### LINKED ISSUE #2782: [Testing] $this->config method
Similarly to how we can set settings to be applied before boot, we could do the same for config values. There's a workaround in place at https://github.com/flarum/core/pull/2732/files#diff-7b87fbe6453c8a913c6a49bd4c7a29321125bb9859374414501c5e83652c6b6dR113-R124, and this could be useful for testing things like the headers PRs.

### PR DESCRIPTION
**Pull request by askvortsov1**
Fix https://github.com/flarum/core/issues/2782

### CODE REVIEW

File: src/integration/TestCase.php [hunk 1]
```diff
@@ -117,7 +124,25 @@ protected function extension(string ...$extensions)
...
+
+    /**
+     * Some Flarum code depends on config.php values. Flarum doesn't
+     * offer a way to set them at runtime, so this method lets you
+     * add/override them before boot.
+     * 
+     * Note that this method will have no effect if called after the
+     * application is booted.
+     */
+    protected function config(string $key, $value)
```
Comment (tankerkiller125) [inline src/integration/TestCase.php:151]: How would we go about setting something like `'headers' => ['referrerPolicy' => 'strict-origin']` with this function. Looking at it right now and the test you've written this example isn't possible.

File: tests/tests/integration/TestCaseTest.php [hunk 2]
```diff
@@ -57,6 +58,31 @@ public function settings_cleaned_up_from_previous_method()
         $this->assertEquals(null, $settings->get('display_name_driver'));
     }
 
+    /**
+     * @test
+     */
+    public function can_add_config_via_method()
+    {
+        $this->config('hello', 'world');
+        $this->config('url', 'https://flarum.org');
```
Comment (tankerkiller125) [inline tests/tests/integration/TestCaseTest.php:67]: Can we get some added test for sub-key config types?

**Review by tankerkiller125 — CHANGES_REQUESTED** [review]

File: src/integration/TestCase.php (hunk 1 above)
Comment (askvortsov1) [reply src/integration/TestCase.php:151]: Use a dot separated syntax. Iirc, shouldn't Arr::set do a deep assignment?

Comment (tankerkiller125) [reply src/integration/TestCase.php:151]: Ah yes you're correct, my quick glance at the code failed to see the other method that actually set's it using `Arr::set()`

**Review by tankerkiller125 — APPROVED** [review]

**Review by SychO9 — APPROVED** [review]

**Pull request merged**